### PR TITLE
Changes made to source and tests to support PHP 7.3

### PR DESCRIPTION
**Pull request by yitam**
See related #804 and #810

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/msphpsql/822)

### CODE REVIEW

File: source/shared/core_sqlsrv.h [hunk 1]
```diff
@@ -2377,10 +2377,13 @@ namespace core {
 
     inline void sqlsrv_array_init( _Inout_ sqlsrv_context& ctx, _Out_ zval* new_array TSRMLS_DC) 
     {
-        int zr = ::array_init(new_array);
```
Comment (david-puglielli) [inline source/shared/core_sqlsrv.h]: Is this necessary? What happens in PHP 7.3 if there is an error with array_init?

File: test/functional/sqlsrv/sqlsrv_errors.phpt [hunk 2]
```diff
@@ -119,7 +119,7 @@ sqlsrv_close returns true even if an error happens.
     echo "Test successfully done.\n";
 ?>
 --EXPECTF--
-Warning: sqlsrv_close() expects parameter 1 to be resource, boolean given in %Ssqlsrv_errors.php on line %x
+Warning: sqlsrv_close() expects parameter 1 to be resource, bool%S given in %Ssqlsrv_errors.php on line %x
```
Comment (david-puglielli) [inline test/functional/sqlsrv/sqlsrv_errors.phpt]: PHP 7.3 changes he output to just `bool` and `int`?

Comment (yitam) [reply test/functional/sqlsrv/sqlsrv_errors.phpt]: yes

File: source/shared/core_sqlsrv.h (hunk 1 above)
Comment (yitam) [reply source/shared/core_sqlsrv.h]: It used to be a function that returns either a SUCCESS or FAILURE but now a macro in 7.3

**Pull request merged**